raft-tech/TANF-app
# [issues/519/backend] Allow creating super users through admin panel

### This pull request changes...

No longer exclude is_superuser and is_staff properties from django admin panel user creation form. In order to use the panel, an initial super user must be created, a migration is added which reads from environment variables to set the first super users credentials.


## FUNCTIONALITY

Addresses issue #519. 

Allow super users to create other super users from the django admin panel.
Add first super user through a migration.

## TO TEST

1. Log in as a user with `staff` and `superuser` flags (If you don't have one, we can make sure you do)
2. Got to https://tdp-backend.app.cloud.gov/admin/
3. Click on `Users`
4. Find your user in the list and click your email address
5. Verify the `Staff` and `Superuser` checkboxes are checked
6. Find a user (someone on the team who can help test) who does not have `Staff` and `Superuser`
7. Open that user's profile and check `Staff` and `Superuser` checkboxes
8. Ask that user to login and access Django Admin
9. Verify they have full access (they can log in and see all of the links on the home page)
10. Go back to that user's profile in your Django Admin
11. Uncheck `Staff` and `Superuser`
12. Ask user to log out and log back in and access Django Admin
13. Verify that they cannot access

### This pull request is ready to merge when...

  - OpenAPI  
  - Readme
  - Entity Relationship Diagram (ERD)

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

- [x] Superuser set via data migration or CF CLI task; demonstrate this locally or in the dev environment to PO and tech lead
- [x] user_role_management.md documentation updated
- [x] ADR created to document this decision

As Product Owner, @lfrohlich will decide if ACs are met.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*


### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*

    + [See the full Accessibility Assessment plan here.](https://github.com/HHS/TANF-app/blob/main/docs/a11y/how-18f-will-test-a11y.md)

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful build with a single command*


### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*


### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*
